src/Playwright/Core/Frame.cs

```diff
@@ -603,6 +598,10 @@ private Task WaitForURLAsync(string urlString, Regex urlRegex, Func<string, bool
         private Waiter SetupNavigationWaiter(string @event, float? timeout)
         {
             var waiter = new Waiter((this.Page as Page)!.Channel, @event);
+            if (this.Page.IsClosed)
```

Comment:
We should refactor it (and in upstream) to only rely on the connection. I.e. don't hide the waitForEventInfo rejection.